### PR TITLE
Remove Cirrus support for Gold

### LINKED ISSUE #67463: Cirrus implementation of Gold needs to be removed
Golden file tests are now running across all platforms on LUCI. 
https://github.com/flutter/flutter/commit/48b9c3d39b00a4b8abba79c4ad1e736bfe447581 added back some framework linux builds for smoke testing, but did not include all of the Gold configs, causing unresolvable failures on PRs making golden file changes. These contributors are blocked currently due to this.

We can solve this easily by just removing the Cirrus implementation from Gold, since it is no longer necessary.
cc/ @godofredoc

### PR DESCRIPTION
**Pull request by Piinks**
## Description

Golden file testing is running across all platforms on LUCI now, making the Cirrus implementation no longer necessary. A few framework builds were added back to Cirrus to serve as smoke tests, but don't need to be running duplicate golden file tests. These builds were added back without the right Gold config, so PRs making a golden file change right now are blocked.

Since we are using LUCI, with this change we will no longer need to handle the special case of first time contributors, so this removes all of the ignore logic from the Gold implementation as well. :) 

TODO:
 - Update the wiki to remove first time contributor guidance since the issue will be resolved

## Related Issues

Fixes https://github.com/flutter/flutter/issues/67463

## Tests

Updated for a Cirrus-free lifestyle. 

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
FYI @kjlubick We can can cross off another manual request point. :)

File: packages/flutter_goldens/lib/flutter_goldens.dart [hunk 1]
```diff
@@ -35,7 +35,7 @@ Future<void> main(FutureOr<void> testMain()) async {
     goldenFileComparator = await FlutterPreSubmitFileComparator.fromDefaultComparator(platform);
   } else if (FlutterSkippingFileComparator.isAvailableForEnvironment(platform)) {
     goldenFileComparator = FlutterSkippingFileComparator.fromDefaultComparator(
-      'Golden file testing is not executed on some Cirrus & Luci environments.'
+      'Golden file testing is not executed on Luci environments outside of flutter/flutter.'
```
Comment (Piinks) [inline packages/flutter_goldens/lib/flutter_goldens.dart:38]:
```suggestion
      'Golden file testing is not executed on Cirrus, or LUCI environments outside of flutter/flutter.'
```

File: packages/flutter_goldens/lib/flutter_goldens.dart [hunk 2]
```diff
@@ -425,74 +345,22 @@ class _AuthorizedFlutterPreSubmitComparator extends FlutterPreSubmitFileComparat
...
+    return luciPreSubmit;
   }
 }
 
 /// A [FlutterGoldenFileComparator] for testing conditions that do not execute
 /// golden file tests.
 ///
-/// Currently, this comparator is used in some Cirrus test shards and Luci
-/// environments.
+/// Currently, this comparator is used in Luci environments when executing tests
```
Comment (Piinks) [inline packages/flutter_goldens/lib/flutter_goldens.dart:362]:
```suggestion
/// Currently, this comparator is used on Cirrus, or in Luci environments when executing tests
```

**Comment by christopherfujino** [comment]
does LUCI get `goldctl` via cipd?

**Comment by Piinks** [comment]
> does LUCI get `goldctl` via cipd?

It does, but that is in the luci recipe itself, I think only cirrus builds get goldctl using the scripts I'm deleting. Here's the luci recipe that ensures goldctl: https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/flutter.py#71

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 6.
View them at https://flutter-gold.skia.org/cl/github/67468

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by xster** [comment]
Thanks for digging through this @Piinks

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
While validating this on the gold dashboard, I noticed there wasn't a windows image. The Windows build here passed, but the logs don't show any tests being run, instead: `ERROR: The process "flutter_tester.exe" not found.`
Filed https://github.com/flutter/flutter/issues/67540

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 7.
View them at https://flutter-gold.skia.org/cl/github/67468

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 8.
View them at https://flutter-gold.skia.org/cl/github/67468

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 9.
View them at https://flutter-gold.skia.org/cl/github/67468

**Comment by Piinks** [comment]
This is blocked on https://github.com/flutter/flutter/issues/67560

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 10.
View them at https://flutter-gold.skia.org/cl/github/67468

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 11.
View them at https://flutter-gold.skia.org/cl/github/67468

**Comment by Piinks** [comment]
This appears to be blocked by a different issue now. The Mac build tests are having version solving issues. I reset my git history with master to see if that would resolve it, but it has not, investigating.

**Pull request merged**